controllers/btpoperator_controller.go

```diff
@@ -504,51 +403,87 @@ func (r *BtpOperatorReconciler) prepareModuleResources(ctx context.Context, us [
 	}
 
 	return nil
 }
 
-func (r *BtpOperatorReconciler) reconcileResources(ctx context.Context, us []*unstructured.Unstructured) error {
-	logger := log.FromContext(ctx)
-	logger.Info("reconciling module resources")
+func (r *BtpOperatorReconciler) addLabels(chartVer string, us ...*unstructured.Unstructured) {
 
 	for _, u := range us {
-		if err := r.Patch(ctx, u, client.Apply, client.ForceOwnership, client.FieldOwner(operatorName)); err != nil {
-			return fmt.Errorf("while reconciling %s %s: %w", u.GetName(), u.GetKind(), err)
+		labels := u.GetLabels()
+		if len(labels) == 0 {
+			labels = make(map[string]string)
 		}
+		labels[managedByLabelKey] = operatorName
+		labels[chartVersionKey] = chartVer
+		u.SetLabels(labels)
 	}
+}
 
-	return nil
+func (r *BtpOperatorReconciler) setNamespace(us ...*unstructured.Unstructured) {
+	for _, u := range us {
+		u.SetNamespace(ChartNamespace)
+	}
 }
 
-func (r *BtpOperatorReconciler) waitForResourcesReadiness(ctx context.Context, us []*unstructured.Unstructured) error {
-	numOfResources := len(us)
-	resourcesReadinessInformer := make(chan bool, numOfResources)
-	allReadyInformer := make(chan bool)
+func (r *BtpOperatorReconciler) deleteCreationTimestamp(us ...*unstructured.Unstructured) {
 	for _, u := range us {
-		go r.checkResourceReadiness(ctx, u, resourcesReadinessInformer)
+		unstructured.RemoveNestedField(u.Object, "metadata", "creationTimestamp")
 	}
-	go func(c chan bool) {
-		timeout := time.After(ReadyTimeout)
-		for i := 0; i < numOfResources; i++ {
-			select {
-			case <-resourcesReadinessInformer:
+}
+
+func (r *BtpOperatorReconciler) setConfigMapValues(secret *corev1.Secret, u *unstructured.Unstructured) error {
+	return unstructured.SetNestedField(u.Object, string(secret.Data["cluster_id"]), "data", "CLUSTER_ID")
+}
+
+func (r *BtpOperatorReconciler) setSecretValues(secret *corev1.Secret, u *unstructured.Unstructured) error {
+	for k := range secret.Data {
+		if err := unstructured.SetNestedField(u.Object, base64.StdEncoding.EncodeToString(secret.Data[k]), "data", k); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *BtpOperatorReconciler) applyResources(ctx context.Context, us []*unstructured.Unstructured) error {
+	for _, u := range us {
+		if err := r.Patch(ctx, u, client.Apply, client.ForceOwnership, client.FieldOwner(operatorName)); err != nil {
+			return fmt.Errorf("while applying %s %s: %w", u.GetName(), u.GetKind(), err)
+		}
+	}
+
+	return nil
+}
+
+func (r *BtpOperatorReconciler) waitForResourcesReadiness(ctx context.Context, us []*unstructured.Unstructured) error {
+	numOfResources := len(us)
+	resourcesReadinessInformer := make(chan bool, numOfResources)
+	allReadyInformer := make(chan bool)
+	for _, u := range us {
+		go r.checkResourceReadiness(ctx, u, resourcesReadinessInformer)
+	}
+	go func(c chan bool) {
+		timeout := time.After(ReadyTimeout)
+		for i := 0; i < numOfResources; i++ {
+			select {
+			case <-resourcesReadinessInformer:
```

Comment:
Why waitGroup is insufficient it this case?

Reply:
WaitGroup doesn't have a timeout